### PR TITLE
User type crud

### PR DESCRIPTION
**Pull request by srkacha**

### CODE REVIEW

File: controller/userTypeController.js [hunk 1]
```diff
@@ -1,7 +1,11 @@
 const UserType = require('../model/userTypeModel');
 const HttpStatus = require('http-status-codes');
 
-const getAll = (req, res) => {
+const create = (req, res) => {
+    const userType = new UserType(req.body);
```
Comment (SavoX2) [inline controller/userTypeController.js]: Is this it? The whole function? O.o

File: controller/userTypeController.js [hunk 2]
```diff
@@ -1,7 +1,11 @@
...
+}
+
+const retrieveAll = (req, res) => {
```
Comment (SavoX2) [inline controller/userTypeController.js]: Check for user is missing.

File: controller/userTypeController.js [hunk 3]
```diff
@@ -22,7 +26,28 @@ const getAll = (req, res) => {
     });
 };
 
+const retrieveById = (req, res) => {
+    const userTypeId = req.body.id;
```
Comment (SavoX2) [inline controller/userTypeController.js]: - `retrieveById` should be a GET request. As such the ID is sent as a URL parameter, not in the body.
- User check is missing.

File: controller/userTypeController.js [hunk 4]
```diff
@@ -22,7 +26,28 @@ const getAll = (req, res) => {
...
+            message: 'There was a problem retrieving the user type.'
+        });
+    });
+}
+
 //Retrieve all user types
 module.exports = {
-    getAll: getAll
+    retrieveAll,
+    retrieveById,
```
Comment (SavoX2) [inline controller/userTypeController.js]: We should discuss internally function naming. For example: you are using `retrieve`, while me and @zirlman are using `get`. We should come to an agreeing how to name them to have a single coding style present in the project.

File: controller/userTypeController.js (hunk 1 above)
Comment (srkacha) [reply controller/userTypeController.js]: No validation because function will not be used.

Comment (srkacha) [reply controller/userTypeController.js]: Also, you were probably looking at an outdated commit

File: controller/userTypeController.js (hunk 2 above)
Comment (srkacha) [reply controller/userTypeController.js]: Added a check for user.

File: controller/userTypeController.js (hunk 3 above)
Comment (srkacha) [reply controller/userTypeController.js]: Outdated commit

File: controller/userTypeController.js (hunk 4 above)
Comment (srkacha) [reply controller/userTypeController.js]: I will rename my functions to match your function names, now and in the future.

File: controller/userTypeController.js [hunk 5]
```diff
@@ -22,7 +77,58 @@ const getAll = (req, res) => {
...
+            userType.value = req.body.value ? req.body.value : userType.value;
+            userType.name = req.body.name ? req.body.name : userType.name;
+            userType.save().then(() => {
+                return res.status(HttpStatus.OK).json(userType);
+            }).catch(err => {
+                const errorMessage = ValidationUtil.buildErrorMessage(err,
+                    'update',
+                    'user type');
+                return res.status(HttpStatus.BAD_REQUEST).json({
+                    status: 'Error',
```
Comment (SavoX2) [inline controller/userTypeController.js]: I know it says it was for practice, but user `StatusEnum` when possible.

File: controller/userTypeController.js [hunk 6]
```diff
@@ -22,7 +77,58 @@ const getAll = (req, res) => {
...
+                    message: errorMessage
+                });
+            })
+        } else {
+            return res.status(HttpStatus.NOT_FOUND).json({
+                status: 'Error',
```
Comment (SavoX2) [inline controller/userTypeController.js]: `StatusEnum`.

**Pull request merged**